Chatkit/Subscriptions/Subscription.swift

```diff
@@ -0,0 +1,301 @@
+import Foundation
+import class PusherPlatform.PPRequestOptions
+import enum PusherPlatform.HTTPMethod
+
+typealias SubscribeHandler = (VoidResult) -> Void
+
+protocol SubscriptionDelegate: AnyObject {
+    func subscription(_ subscription: Subscription, didReceiveEventWithJsonData jsonData: Data)
+    func subscription(_ subscription: Subscription, didReceiveError error: Error)
+}
+
+protocol Subscription {
+    func subscribe(completion: @escaping SubscribeHandler)
+    func unsubscribe()
+}
+
+class ConcreteSubscription: Subscription {
+    
+    enum State {
+        case notSubscribed
+        case subscribingStageOne(instanceWrapper: InstanceWrapper, completions: [SubscribeHandler])
+        case subscribingStageTwo(instanceWrapper: InstanceWrapper, resumableSubscription: ResumableSubscription, completions: [SubscribeHandler])
+        case subscribed(instanceWrapper: InstanceWrapper, resumableSubscription: ResumableSubscription)
+    }
+    
+    typealias Dependencies = HasInstanceWrapperFactory
+    
+    let subscriptionType: SubscriptionType // Internal `get` aids testing
+    private let dependencies: Dependencies
+    weak var delegate: SubscriptionDelegate?
+    
+    private(set) var state: State = .notSubscribed // Internal `get` aids testing
```

Comment:
The same applies here.

Reply:
I *do* completely agree with the sentiment.  If I remove this access however the value/quality of the unit tests for this class will drop significantly.  Personally I consider the security and confidence that the unit tests currently provide more valuable than this small violation to an *internal* interface.  If you feel strongly however then I am happy to remove it, just let me know.

Reply:
Let's leave it as it is.